docs/advanced/tls_porting.md

```diff
@@ -14,61 +14,77 @@ Creating session keys is only one use for random values; they have far more comp
+```C
+int trng_get_bytes(trng_t *obj,
+                   uint8_t *output, size_t length, size_t *output_length);
+```
 
-- ``unsigned char *output``: A pointer to the output buffer. The function should write the entropy it collected to the buffer; mbed TLS then uses this data as entropy. Please consult your board's manual and write only the strongest entropy possible in this buffer. 
+- ``trng_t *obj``: `trng_t` is an alias to `trng_s` and it is the callers responsibility to initialise it before passing it to this function and deinitialise it (with the help of `trng_init()` and `trng_free()` respectively) when it is not required anymore.
 
-	**Warning:** Although it is possible to fill this buffer without a strong hardware entropy source, we strongly advise against it, because it will nullify any security provided by mbed TLS.
+- ``uint8_t *output``: A pointer to the output buffer. The function should write the entropy it collected to the buffer; mbed TLS then uses this data as entropy. Please consult your board's manual and write only the strongest entropy possible in this buffer.
 
-- ``size_t len``: The length of the output buffer. The function shouldn't write more data than this to the output buffer under any circumstances.
+    **Warning:** Although it is possible to fill this buffer without a strong hardware entropy source, we strongly advise against it, because it will nullify any security provided by mbed TLS.
 
-- ``size_t *olen``: The length of the data written into the output buffer. It tells the caller how much entropy has been collected and how many bytes of the output buffer it can use. It should always reflect the exact amount of entropy collected; setting it higher than the actual number of bytes collected is a serious security risk.
+- ``size_t length``: The length of the output buffer. The function shouldn't write more data than this to the output buffer under any circumstances.
 
-### Setting the macros
+- ``size_t *output_length``: The length of the data written into the output buffer. It tells the caller how much entropy has been collected and how many bytes of the output buffer it can use. It should always reflect the exact amount of entropy collected; setting it higher than the actual number of bytes collected is a serious security risk.
 
-To register your `mbedtls_hardware_poll` function with the mbed TLS entropy framework, you have to add `MBEDTLS_ENTROPY_HARDWARE_ALT` to your macros in `targets.json`:
 
-```
-"macros": ["MBEDTLS_ENTROPY_HARDWARE_ALT", etc.]
-```
+### Indicating the presence of a TRNG
 
-You can read more about how to add a macro for your target [here](../mbed_OS/Targets.md).
+To indicate that the target has an entropy source, you have to add `TRNG` to the capabilities of the target in `targets.json`:
```

Comment:
I deleted the link because the document it was pointing to does not exist anymore.